### DISCUSSION TITLE
Adding message size limit in receive function pgproto3

### DISCUSSION THREAD

**Discussion by Jrmy2402** [jackc/pgx#1836] [Ideas]
Hello,

I would like to propose the addition of a feature to limit the size of messages received by the `Backend.Receive()` function. Specifically, it would be useful to set a maximum message size parameter to prevent the processing of unusually large messages that might impact performance or pose security risks.

In certain situations, especially when dealing with external or untrusted data sources, there is a potential risk of receiving and processing very large messages, which could lead to performance degradation or exploitation through buffer overflow or similar attacks.

If this proposal aligns with your vision for `pgx`, I'd be more than willing to contribute by opening a Merge Request to implement this feature.

Thank you for considering this proposal, and for your continuous efforts in maintaining and improving `pgx`.

**Comment by jackc** [comment]
I'm open to the _idea_, but I'm not sure exactly how it would be used or what the interface would look like.

Extremely large messages are possible. PostgreSQL allows fields of up to 1 GB. So presumably a single DataRow could be at least that big.

**Reply by Jrmy2402** [reply]
Ok, good. We will open an MR.
Yes, PostgreSQL allows large messages but in our case, we want to limit the size to avoid overflow memory in our proxy.
This will be optional, so each one can choose the size.

**Reply by Jrmy2402** [reply]
I created the pull request https://github.com/jackc/pgx/pull/1839 to add the feature. 
Please, tell me if it seems ok for you or not.

**Reply by jackc** [reply]
Merged

**Reply by Jrmy2402** [reply]
Hi, when do you forecast creating a new release of PGX so I can use the tag instead of the master branch?
Thanks in advance

**Reply by Jrmy2402** [reply]
Hi, I have pushed a new commit to expose MaxExpectedBodyLen and ActualBodyLen in ExceededMaxBodyLenErr struct.
https://github.com/jackc/pgx/pull/1862
Please, tell me if it seems ok for you or not.

**Reply by jackc** [reply]
Merged.
